cg/meta/demultiplex/utils.py

```diff
@@ -74,39 +74,19 @@ def create_delivery_file_in_flow_cell_directory(flow_cell_directory: Path) -> No
     Path(flow_cell_directory, DemultiplexingDirsAndFiles.DELIVERY).touch()
 
 
-def get_sample_internal_ids_from_flow_cell(flow_cell_data: FlowCellDirectoryData) -> List[str]:
-    samples: List[FlowCellSample] = flow_cell_data.get_sample_sheet().samples
-    return get_valid_sample_ids(samples)
-
-
-def get_valid_sample_ids(samples: List[FlowCellSample]) -> List[str]:
-    """Get all valid sample ids from sample sheet."""
-    valid_sample_ids = [
-        sample.sample_id for sample in samples if is_valid_sample_id(sample.sample_id)
-    ]
-    formatted_sample_ids = [sample_id_index.split("_")[0] for sample_id_index in valid_sample_ids]
-    return formatted_sample_ids
-
-
 def get_q30_threshold(sequencer_type: Sequencers) -> int:
     return FLOWCELL_Q30_THRESHOLD[sequencer_type]
 
 
-def get_sample_sheet_path(flow_cell_directory: Path) -> Path:
-    """
-    Recursively searches for the given sample sheet file in the provided flow cell directory.
-
-    Raises:
-        FileNotFoundError: If the sample sheet file is not found in the flow cell directory.
-    """
-    for directory_path, _, files in os.walk(flow_cell_directory):
-        if DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME in files:
-            LOG.info(f"Found sample sheet in {directory_path}")
-            return Path(directory_path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME)
-
-    raise FileNotFoundError(
-        f"Sample sheet not found in given flow cell directory: {flow_cell_directory}"
+def get_sample_sheet_path(
+    flow_cell_directory: Path,
+    sample_sheet_file_name: str = DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
+) -> Path:
+    """Get the path to the sample sheet in the flow cell directory."""
+    sample_sheet_path: Path = get_file_in_directory(
+        directory=flow_cell_directory, file_name=sample_sheet_file_name
     )
+    return sample_sheet_path
```

Comment:
Can be returned immediately
```suggestion
    return get_file_in_directory(
        directory=flow_cell_directory, file_name=sample_sheet_file_name
    )
```